lib/kitchen/transport/winrm.rb

```diff
@@ -38,22 +36,36 @@ class WinrmFailed < TransportFailed; end
     # @author Salim Afiune <salim@afiunemaya.com.mx>
     # @author Fletcher Nichol <fnichol@nichol.ca>
     class Winrm < Kitchen::Transport::Base
-
       kitchen_transport_api_version 1
 
       plugin_version Kitchen::VERSION
 
-      default_config :port, 5985
       default_config :username, "administrator"
       default_config :password, nil
-      default_config :endpoint_template, "http://%{hostname}:%{port}/wsman"
       default_config :rdp_port, 3389
       default_config :connection_retries, 5
       default_config :connection_retry_sleep, 1
       default_config :max_wait_until_ready, 600
+      default_config :winrm_transport
+      default_config :port do |transport|
+        transport[:winrm_transport] == :ssl ? 5986 : 5985
+      end
+      default_config :endpoint_template do |transport|
+        scheme = transport[:winrm_transport] == :ssl ? "https" : "http"
+        "#{scheme}://%{hostname}:%{port}/wsman"
+      end
 
       # (see Base#connection)
       def connection(state, &block)
+        case config[:winrm_transport]
+        when "ssl"
+          config[:winrm_transport] = :ssl
+        when "sspinegotiate" && host_os_windows?
+          config[:winrm_transport] = :sspinegotiate
+        else
+          config[:winrm_transport] = :plaintext
+        end
```

Comment:
This block would work better in a `def finalize_config!(instance)` method, similar to https://github.com/test-kitchen/test-kitchen/blob/621cd45784c3aed0e893d390c2758125b250151a/lib/kitchen/transport/ssh.rb#L62-L75 (note the `super` call there is pretty important :smile:)

The reason is that `kitchen diagnose` returns its configuration after the `#finalize_config!` has done any last minute tweaks, meaning that the final computed value of `config[:winrm_transport]` would show in the diagnostic output.